openrouteservice/src/main/java/org/heigit/ors/matrix/MultiTreeMetricsExtractor.java

```diff
@@ -139,13 +145,19 @@ public void calcValues(MultiTreeSPEntry[] targets, MatrixLocations srcData, Matr
 					MultiTreeSPEntry targetEntry = targets[i];
 
 					if (targetEntry != null) {
-						pathTime = 0.0;
-						pathDistance = 0.0;
-						pathWeight = 0.0;
+						//Only set values to 0 if target and start node are the same
+						if(srcData.getNodeId(j) == targetEntry.getAdjNode()) {
+							pathTime = 0.0;
+							pathDistance = 0.0;
+							pathWeight = 0.0;
+						}
```

Comment:
Code duplication of 
```
pathTime = 0.0;
pathDistance = 0.0;
pathWeight = 0.0;
```
could be avoided by modifying the condition in L157 to be `if (sptItem.getParent() != null || srcData.getNodeId(j) == targetEntry.getAdjNode()) as in the case when start and target nodes are the same the subsequent while loop won't execute anyway.

Reply:
This would execute the following code for `srcData.getNodeId(j) == targetEntry.getAdjNode() && sptItem.getParent() == null`?

Reply:
I'm sorry, I don't follow. My point was that 
```
						//Only set values to 0 if target and start node are the same
						if(srcData.getNodeId(j) == targetEntry.getAdjNode()) {
							pathTime = 0.0;
							pathDistance = 0.0;
							pathWeight = 0.0;
						}
```
is probably obsolete as this special case could be covered by the extended condition in L157, or do I miss sth?

Reply:
If I combine the two conditions in one (combined in L157), then I have the possibility of executing the code from L161 onwards even if `sptItem.getParent() == null`

Reply:
I have rewritten the first condition.

Reply:
> If I combine the two conditions in one (combined in L157), then I have the possibility of executing the code from L161 onwards even if `sptItem.getParent() == null`

Of course, you are right, I didn't read carefully. Thanks!